### DISCUSSION TITLE
a message serialization support

### DISCUSSION THREAD

**Discussion by wuyueandrew** [alibaba/GraphScope#2595] [General]
**Is your feature request related to a problem? Please describe.**
My cpp_pregel algo needs to send a structed message and now it is not supported.
My solution is connecting these field using concat, it's not elegant.

**Describe the solution you'd like**
a specify message serialization , like flatbuffers which can offer a encapsulation of structed message.

**Comment by yecol** [comment]
You may want to implement the serialization/deserialization for your structure in `InArchive`/`OutArchive`..
